sdk/index.ts

```diff
@@ -0,0 +1,273 @@
+            );
+        }
+
+        return data.json();
+    }
+
+    private async getEnvironmentFromApi() {
+        const environment_data = await this.getJSONResponse(this.environmentUrl, 'GET');
+        return buildEnvironmentModel(environment_data);
+    }
+
+    private getEnvironmentFlagsFromDocument() {
+        if (!this.environment) {
+            throw new FlagsmithClientError(
+                'Unable to build identity model when no local environment present.'
+            );
+        }
+
+        return Flags.fromFeatureStateModels({
+            featureStates: getEnvironmentFeatureStates(this.environment),
+            analyticsProcessor: this.analyticsProcessor,
+            defaultFlagHandler: this.defaultFlagHandler
+        });
+    }
+
+    private getIdentityFlagsFromDocument(identifier: string, traits: { [key: string]: any }) {
+        if (!this.environment) {
+            throw new FlagsmithClientError(
+                'Unable to build identity model when no local environment present.'
+            );
+        }
+
+        const identityModel = this.buildIdentityModel(identifier, Object.values(traits));
+        const featureStates = getIdentityFeatureStates(this.environment, identityModel);
+
+        return Flags.fromFeatureStateModels({
+            featureStates: featureStates,
+            analyticsProcessor: this.analyticsProcessor,
+            defaultFlagHandler: this.defaultFlagHandler
+        });
+    }
+
+    private async getEnvironmentFlagsFromApi() {
+        try {
+            const apiFlags = await this.getJSONResponse(this.environmentFlagsUrl, 'GET');
+            return Flags.fromAPIFlags({
+                apiFlags: apiFlags,
+                analyticsProcessor: this.analyticsProcessor,
+                defaultFlagHandler: this.defaultFlagHandler
+            });
+        } catch (e) {
+            if (this.defaultFlagHandler) {
+                return new Flags({
+                    flags: {},
+                    defaultFlagHandler: this.defaultFlagHandler
+                });
+            }
+
+            throw e;
+        }
+    }
+
+    private async getIdentityFlagsFromApi(identifier: string, traits: { [key: string]: any }) {
+        try {
+            const data = generateIdentitiesData(identifier, traits);
+            const jsonResponse = await this.getJSONResponse(this.identitiesUrl, 'POST', data);
+            return Flags.fromAPIFlags({
+                apiFlags: jsonResponse['flags'],
+                analyticsProcessor: this.analyticsProcessor,
+                defaultFlagHandler: this.defaultFlagHandler
+            });
+        } catch (e) {
+            if (this.defaultFlagHandler) {
+                return new Flags({
+                    flags: {},
+                    defaultFlagHandler: this.defaultFlagHandler
+                });
+            }
+
+            throw e;
+        }
+    }
+
+    private buildIdentityModel(identifier: string, traits: any[]) {
```

Comment:
This traits: any[] was the cause of not catching https://github.com/Flagsmith/flagsmith-nodejs-client/pull/23/files#diff-66fd8b30e073dfc58786b90e78803df61c90731cd651e24cce5163022f08e43fR208